code/modules/shuttle/on_move.dm

```diff
@@ -331,6 +331,18 @@ All ShuttleMove procs go here
 	if(. & MOVE_AREA)
 		. |= MOVE_CONTENTS
 
+/obj/structure/ladder/afterShuttleMove(turf/oldT, list/movement_force, shuttle_dir, shuttle_preferred_direction, move_dir, rotation)
+	. = ..()
+	// be moved, but re-connect at the destination
+	disconnect()
```

Comment:
You should probably disconnect in beforeShuttleMove, afterShuttleMove gets delayed by check ticks and it's potentially possible to use the ladder before it's been updated to the new location.